README.md

```diff
@@ -17,3 +17,380 @@ Implement a prototype job worker service that provides an API to run arbitrary L
 
 * Client should be able to query result of the job execution and fetch the logs
 
+## Design Document
+
+### Limitations And Scope
+
+#### Data Management
+Though the use of a database to store persistant data would be ideal, I will be instead storing the outputs and error outputs into logs stored on 
+the file system of the linux worker. The logs will be generated at start time with the foldername \<uuid\>-\<startTimeStamp\>
+The client can query the server to see what is running to determine what can be killed and a query a list of jobs that were executed,
+or the client can store response data from server.
+
+#### Scale
+The scope of this project would only deal with a single linux worker server interfacing with multiple clients
+
+### API Design
+
+* The client will have an Command Api that takes in three types of request, Start, Stop, and Query
+
+* The client can then execute the commands over the server via Execute APIs'
+
+#### Start
+
+* The Start command is called with a StartRequest that has the client's command and required arguments and optional env, dir params
+
+* A uuid (universal unique identification) will be generated and a folder called START-\<startTimeStamp\>will be created, two logs called PID-<pid>-stdout.log and PID-<pid>-stderr.log will be created
```

Comment:
no need to include `PID` in log file names, it will complicate the implementation.
the server can store the PID in its memory, the client should only care about the UUID